packages/hierarchies/src/hierarchies/imodel/FilteringHierarchyDefinition.ts

```diff
@@ -109,248 +111,149 @@ export class FilteringHierarchyDefinition implements HierarchyDefinition {
 
   public get parseNode(): NodeParser {
     return async (row: { [columnName: string]: any }, parentNode?: HierarchyDefinitionParentNode): Promise<SourceInstanceHierarchyNode> => {
-      const hasFilterTargetAncestor: boolean = !!row[ECSQL_COLUMN_NAME_HasFilterTargetAncestor];
+      const parsedNode = await (this._source.parseNode ?? defaultNodesParser)(row);
 
-      const { filteredChildrenIdentifierPaths, filterTargetOptions, isFilterTarget } =
+      const filteringHelper = createHierarchyFilteringHelper(this._nodeIdentifierPaths, parentNode);
+      const nodeExtraProps =
         row[ECSQL_COLUMN_NAME_FilterECInstanceId] && row[ECSQL_COLUMN_NAME_FilterClassName]
-          ? await this.getChildrenIdentifierPaths(parentNode?.filtering?.filteredChildrenIdentifierPaths ?? this._nodeIdentifierPaths, {
-              className: row[ECSQL_COLUMN_NAME_FilterClassName],
-              id: row[ECSQL_COLUMN_NAME_FilterECInstanceId],
-            })
-          : { filteredChildrenIdentifierPaths: [], filterTargetOptions: undefined, isFilterTarget: false };
-
-      const defaultNode = await (this._source.parseNode ?? defaultNodesParser)(row);
-      return applyFilterAttributes({
-        node: defaultNode,
-        filteredChildrenIdentifierPaths,
-        isFilterTarget,
-        filterTargetOptions,
-        hasFilterTargetAncestor,
-      });
-    };
-  }
-
-  private async getChildrenIdentifierPaths(filteredChildrenNodeIdentifierPaths: HierarchyFilteringPath[], providedIdentifier: InstanceKey) {
-    const { id, className } = providedIdentifier;
-
-    let filteredChildrenIdentifierPaths: HierarchyFilteringPath[] | undefined;
-    let isFilterTarget = false;
-    let filterTargetOptions: HierarchyFilteringPathOptions | undefined;
-    for (const filteredChildrenNodeIdentifierPath of filteredChildrenNodeIdentifierPaths) {
-      const { path, options } = HierarchyFilteringPath.normalize(filteredChildrenNodeIdentifierPath);
-      /* c8 ignore next 3 */
-      if (path.length === 0) {
-        continue;
+          ? await (async () => {
+              const rowInstanceKey = { className: row[ECSQL_COLUMN_NAME_FilterClassName], id: row[ECSQL_COLUMN_NAME_FilterECInstanceId] };
+              return filteringHelper.createChildNodePropsAsync({
+                // eslint-disable-next-line @typescript-eslint/promise-function-async
+                pathMatcher: (identifier) => {
```

Comment:
Can't we do this:
`pathMatcher: (identifier): boolean | Promise<boolean> => {`
instead of adding ignore comment?

Reply:
Yes, that's cleaner. Changed.